auth.c

```diff
@@ -419,8 +419,12 @@ getpwnamallow(struct ssh *ssh, const char *user)
 	struct connection_info *ci;
 	u_int i;
 
+	if ((pw = getpwnam(user)) != NULL)
+		pw = pwcopy(pw);
```

Comment:
the call to pwcopy should stay at the end, otherwise you have to free the copy
in all other 'return null' cases.

Reply:
yes, but see below